librosa/util/_nnls.py

```diff
@@ -63,11 +64,11 @@ def _nnls_lbfgs_block(A, B, x_init=None, **kwargs):
     # If we don't have an initial point, start at the projected
     # least squares solution
     if x_init is None:
-        x_init = np.linalg.lstsq(A, B, rcond=None)[0]
+        x_init = np.einsum('fm,...mt->...ft', np.linalg.pinv(A), B, optimize=True)
         np.clip(x_init, 0, None, out=x_init)
 
     # Adapt the hessian approximation to the dimension of the problem
-    kwargs.setdefault("m", A.shape[1])
+    kwargs.setdefault("m", A.shape[-1])
```

Comment:
We don't need to use -1 here. It's okay to restrict to A being a 2d array (matrix) here.  Only B needs to support multichannel.